### ISSUE TITLE
Issue 03: non-finite MAE loss 発生時の diagnostic JSON を出力する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#472]
# Issue 03: non-finite MAE loss 発生時の diagnostic JSON を出力する

## 背景

現在 `src/seis_attr_ssl/training/mae.py` は non-finite loss を検出すると次の例外で停止する。

```text
non-finite MAE loss at epoch ..., step ...
```

これだけでは原因 sample を追えない。Phase 7.5 では、non-finite 発生時に survey、crop 座標、attribute ids、loss component、input/target/context の finite check と統計量を JSON として保存する。

## 目的

non-finite MAE loss 発生時に、再現と除外判断に必要な診断情報を `<output_root>/diagnostics/` 以下へ保存する。

## 依存 issue

なし。Issue 01/02 とは独立に進められる。

## 変更対象

- 追加: `src/seis_attr_ssl/training/diagnostics.py`
- 更新: `src/seis_attr_ssl/training/mae.py`
- 更新: `src/seis_attr_ssl/training/__init__.py` は必要なら public export
- 追加または更新: `tests/test_mae_training_engine.py`

### diagnostic writer

`src/seis_attr_ssl/training/diagnostics.py` に、JSON-safe な tensor summary helper を追加する。

推奨 API:

```python
def summarize_tensor(value: object) -> dict[str, object]: ...

def summarize_loss_components(losses: Mapping[str, torch.Tensor]) -> dict[str, object]: ...

def build_mae_nonfinite_diagnostic(
    *,
    global_step: int,
    epoch: int,
    batch_index: int,
    batch: Mapping[str, object],
    output: Mapping[str, object],
    losses: Mapping[str, torch.Tensor],
) -> dict[str, object]: ...

def write_json_diagnostic(payload: Mapping[str, object], path: str | Path) -> Path: ...
```

`json.dumps(..., allow_nan=False)` で書けること。loss や tensor 統計が NaN/Inf の場合は、その float を直接 JSON に入れず、次のような形式にする。

```json
{
  "value": null,
  "finite": false,
  "repr": "nan"
}
```

### 必須 diagnostic fields

non-finite loss 発生時に最低限以下を出す。

```text
global_step
epoch
batch_index
survey_id
local_start_xyz
local_compute_start_xyz
context_compute_start_xyz
attribute_ids
loss component values
x/target/context finite check
x/target/context min/max/mean
```

加えて、原因切り分けのため次も入れる。

- `pred_patches` finite check / min / max / mean
- `target_valid` の true count / false count
- `spatial_mask` の true count / false count
- `dropped_attribute_mask` の true count / false count
- `context_valid_mask` がある場合は true count / false count
- `torch_amp_enabled`

### tensor summary 仕様

float tensor の summary は以下を含める。

```json
{
  "present": true,
  "dtype": "torch.float32",
  "shape": [1, 4, 128, 128, 128],
  "numel": 8388608,
  "finite_count": 8388608,
  "nan_count": 0,
  "posinf_count": 0,
  "neginf_count": 0,
  "all_finite": true,
  "min": -3.2,
  "max": 4.1,
  "mean": 0.02
}
```

boolean / integer mask の summary は true/false count または min/max を出す。巨大 tensor 全体を `.tolist()` しないこと。

### training loop 統合

`train_mae_one_epoch()` に以下の optional 引数を追加する。

```python
diagnostics_dir: Path | None = None
```

`run_mae_pretraining()` では `paths.output_root` と `train.diagnostics_dir` から diagnostic 出力先を解決する。

- `train.diagnostics_dir` が未指定なら `<output_root>/diagnostics`
- 相対パスなら `<output_root> / train.diagnostics_dir`
- 絶対パスならそのまま使う

non-finite loss を検出したら、以下のファイル名で保存する。

```text
<diagnostics_dir>/nonfinite_mae_step_<global_step:08d>.json
```

保存後、`FloatingPointError` を raise する。例外メッセージには diagnostic path を含める。

```text
non-finite MAE loss at epoch 3, step 1042; diagnostic written to .../nonfinite_mae_step_00001042.json
```

## テスト要件

`tests/test_mae_training_engine.py` に追加するか、専用 test file を作る。

1. `mae_pretraining_loss` を monkeypatch して `loss = torch.tensor(float('nan'))` を返す。
2. `train_mae_one_epoch()` が `FloatingPointError` を raise する。
3. diagnostic JSON が書かれる。
4. JSON に `global_step`, `epoch`, `batch_index`, `coords`, `attribute_ids`, `losses`, `tensors.x`, `tensors.target`, `tensors.context`, `tensors.pred_patches` が含まれる。
5. JSON は `json.loads()` で読め、標準 JSON として NaN/Inf literal を含まない。
6. `coords` は `mae_collate_fn()` 後の list-of-dict 形式から sample ごとに出る。

## 完了条件

- non-finite loss 発生時に原因 sample を survey_id と crop 座標まで追える。
- diagnostic 出力は巨大 tensor の raw dump を含まない。
- diagnostic 出力後に既存通り training は停止する。

## 実行コマンド

```bash
python -m pytest tests/test_mae_training_engine.py
python -m pytest tests/test_mae_collate.py
```

## スコープ外

- gradient clipping。これは Issue 04。
- checkpoint resume。これは Issue 06/07。
